cmd/metacache-server-pool.go

```diff
@@ -238,7 +238,7 @@ func (z *erasureServerPools) listPath(ctx context.Context, o *listPathOptions) (
 	}(*o)
 
 	entries, err = filteredResults()
-	cancelList()
+	defer cancelList()
```

Comment:
When filteredResults() returns we supposedly have the results we want, and we can cancel. 

Deferring this means the list will run to completion, wasting a lot of resources before we return.

This is definitely not the solution you are looking for,